internal/api/preview_config.go

```diff
@@ -1,19 +1,26 @@
 package api
 
 type PreviewConfig struct {
-	ProjectSlug   string                 `json:"projectSlug"`
-	Credentials   map[string]string      `json:"credentials"`
-	PackageParams map[string]interface{} `json:"packageParams"`
+	ProjectSlug string                    `json:"projectSlug"`
+	Credentials []Credential              `json:"credentials"`
+	Packages    map[string]PreviewPackage `json:"packages"`
+}
+
+type PreviewPackage struct {
+	Params           map[string]interface{} `json:"params"`
+	Secrets          []Secret               `json:"secrets,omitempty"`
+	RemoteReferences []RemoteRef            `json:"remoteReferences,omitempty"`
```

Comment:
Note these are all maps now, not lists.

See https://github.com/massdriver-cloud/massdriver/pull/2047.